### PR TITLE
SHEL-1249: samples: wifi: scan: Fix Kconfig options

### PR DESCRIPTION
**Pull request by D-Triveni**
Removed unnecessary Kconfig options from Scan sample

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

### CODE REVIEW

File: samples/wifi/scan/prj.conf [hunk 1]
```diff
@@ -8,28 +8,18 @@ CONFIG_WIFI_NRF700X=y
...
 CONFIG_NET_L2_ETHERNET=y
-# TODO: Debug why DHCP is needed to scan
-CONFIG_NET_DHCPV4=y
-CONFIG_NET_IPV4=y
-CONFIG_NET_UDP=y
 
 CONFIG_INIT_STACKS=y
 
 # Memories
 CONFIG_MAIN_STACK_SIZE=4096
```
Comment (krish2718) [inline samples/wifi/scan/prj.conf]: can you check if this can be reduced?

File: samples/wifi/scan/prj.conf [hunk 2]
```diff
@@ -8,28 +8,18 @@ CONFIG_WIFI_NRF700X=y
 CONFIG_NET_L2_WIFI_MGMT=y
 CONFIG_HEAP_MEM_POOL_SIZE=25000
 
-CONFIG_NET_PKT_RX_COUNT=1
-CONFIG_NET_PKT_TX_COUNT=1
-
 # System settings
 CONFIG_NEWLIB_LIBC=y
 CONFIG_NEWLIB_LIBC_NANO=n
```
Comment (krish2718) [inline samples/wifi/scan/prj.conf]: I guess without the supplicant we can get this working with nano, its much smaller form factor.

File: samples/wifi/scan/prj.conf [hunk 3]
```diff
@@ -8,28 +8,18 @@ CONFIG_WIFI_NRF700X=y
...
 
 # Networking
 CONFIG_NETWORKING=y
```
Comment (krish2718) [inline samples/wifi/scan/prj.conf:15]: you don't need the full networking stack, so, CONFIG_NET_NATIVE=n should do

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/prj.conf (hunk 1 above)
Comment (D-Triveni) [reply samples/wifi/scan/prj.conf]: > can you check if this can be reduced?

Can be reduced to 2048.

File: samples/wifi/scan/prj.conf (hunk 3 above)
Comment (D-Triveni) [reply samples/wifi/scan/prj.conf:15]: Seeing compilation issues after adding CONFIG_NET_NATIVE kconfig. So once this is fixed in driver, will add.

**Review by krish2718 — APPROVED** [review]
Commit log can be:

```
Remove unnecessary Kconfig options to reduce memory.

Observed the Flash and RAM usage reduction by approx. 
55 KB and 10KB respectively.
```

**Comment by D-Triveni** [comment]
As changes are finalized, closing.

**Pull request closed without merging**